### PR TITLE
new package ready

### PR DESCRIPTION
**Pull request by dg-shadow**

### CODE REVIEW

**Comment by ugocupcic** [comment]
Can you add a link to this package in the main README?

**Comment by dg-shadow** [comment]
Done

On 11 February 2016 at 14:51, Ugo Cupcic notifications@github.com wrote:

> Can you add a link to this package in the main README?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/shadow-robot/sr-config/pull/59#issuecomment-182895575
> .

## 

_Dan Greenwald_
Software Engineer

+44 (0) 20 3051 8149 ext 316
dg@shadowrobot.com

## 

---

_Shadow Robot Company Ltd._
251 Liverpool Road, N1 1LX, UK

## Registered Number 3308007 (England & Wales)

File: sr_multi_camera_config/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Extrinsic Calibration of Multiple Cameras
```
Comment (ugocupcic) [inline sr_multi_camera_config/README.md:1]: Can you add some 

```
roslaunch toto toto.launch
```

then 

```
rosservice call tutu
```

to have an easier to (not) read readme?

File: sr_multi_camera_config/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+This package contains the bare-bones of what is necessary to use the industrial_extrinsic_calibration stack to "automatically" calibrate a scene involving multiple cameras.
+## Initial Configuration
+Principally, there are three things that need to be configured.
+
+### Target
+#### Frame
+The location of the target is set via the link "target_frame" and the joint which connects it to the world_frame in camera_scene.xacro. This file is found in the urdf directory of this package.
+#### Setup
+camera_scene_targets.yaml in the yaml directory of this package defines the target(s) we will use, and their location within the target_frame. The best place to look for these are in (industrial_extrinsic_cal)/targets, where you can find both the images from which to make the physical targets, and their corresponding yaml descriptions.
```
Comment (ugocupcic) [inline sr_multi_camera_config/README.md]: Transform `(industrial_extrinsic_cal)/targets` into a link

File: sr_multi_camera_config/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+### Camera setup
+camera_scene_cameras.yaml in the yaml directory of this package is where we define the cameras whose positions we will be calibrating. The main thing is to make sure the images topics correspond to the setup of the camera driver, and the frames correspond to those defined in camera_scene.xacro. At the moment, they're all called "asus" as that's how they were named in the example.
+
+### Calibration Job
+camera_scene_caljob.yaml is where we assocaiate each camera with the target with which it will be calibrated. If a camera can see multiple targets, you can limit the range of interest (in pixels) so that only the desired target is selected.
```
Comment (ugocupcic) [inline sr_multi_camera_config/README.md]: typo in associate

File: sr_multi_camera_config/README.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+## Calibrating
+First launch you cameras. The file "cameras.launch" is included to launch two of the standard Logitech web-cams we use at shadow. Remeber to set the camera serial numbers to reflect the particular cameras you have.
```
Comment (ugocupcic) [inline sr_multi_camera_config/README.md]: typo in remember

File: sr_multi_camera_config/README.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+Then launch the calibration job - camera_scene_cal.launch will do the trick.
+
+Finally, call the service with ```rosservice call /calibration_service "{allowable_cost_per_observation: 0.25}"``` The allowable_cost_per_observation sets how accurate the calibration must be to be accepted. 0.25 is the default, and it seems to work well enough. Maybe we should experiment with this a bit.
+
+The output of the calibration is a new yaml file called "mutable_joint_states.yamlnew". Rename this to just ".yaml" to use the new calibration.
+
+## Using the Calibration
+Launching publish_tf.launch will look at the xacro and relevenat yaml files and publish the tfs needed to use the new calibration. Running ```gen_urdf.sh``` in the urdf directory will generate a fixed urdf of the camera setup for using later.
```
Comment (ugocupcic) [inline sr_multi_camera_config/README.md]: relevant

**Pull request merged**